README.md

```diff
@@ -1,69 +1,186 @@
+
+The token plugins standard is designed to be secure and to prevent asset loss, gas, and DoS attacks on transfers.
+
+## Primary Benefits
+
+- **100% permissionless from the token contract owner**: Open to all participants.
+- **Risk-free participation**: Token plugins do not require any approval, deposit, or transfer of funds into an external contract for participation.
+- **Multiple plugin connections**: Users can connect with multiple plugins, allowing for simultaneous involvement in multiple incentive programs or governance systems, etc. (subject to a predefined limit, set at deployment).
+- **Simple to adopt**: Implementation is only 150 lines of code.
+- **High security**: Audited by top-notch firms such as OpenZeppelin, providing the highest standard of security.
+- **Built-in reentrancy protection**: [ReentrancyGuard.sol](https://github.com/1inch/token-plugins/blob/master/contracts/libs/ReentrancyGuard.sol) is integrated.
+- **Custom ERC20 representation**: A plugin can be represented by its own associated ERC20 (custom inheritance), enabling a participant to receive benefits for simply holding the token.
+- **Future support**: NFT (ERC721) and Multi-token (ERC1155) support are coming soon!
+
+## Use-Cases
+
+Here are some examples of how Token Plugins is currently being (or could be used) today:
+
+### st1INCH resolver delegation
+Staked 1INCH tokens earn rewards from Resolvers in the Intent Swap system. In order to earn rewards, the tokens must be pointed to a specific Resolver. The resolver is incentivized to have st1INCH pointed at them, so they will reward delegators with some amount of funds. The delegation of st1INCH is done with a token plugin, so there is no need to transfer the tokens to another contract.
+
+### Weighted voting power
+VE governance models like veCRV require the user to lock tokens for a certain amount of time to earn voting rights. This signals to the protocol a long-term vested interest and greatly reduces the surface area for governance attacks. With Token Plugins, this model can now be reversed and align long-term token holders with additional voting power.
+
+### LP-Token farming
+Some protocols incentivize LP token holders with additional rewards beyond swap fees through an additional yield contract that holds the LP tokens and distributes the rewards proportionally to the participating LPs. With token plugins, this extra rewards for LP holders can continue to be opt-in without the need to deposit those LP tokens into a secondary contract. (See [1inch Fusion pods](https://etherscan.io/address/0x806d9073136c8A4A3fD21E0e708a9e17C87129e8#code))
+
+### Shadow staking
+If a protocol wanted to simply reward holders of their token, they could reward them similarly to the weighted voting power method, but instead of increasing voting power over time, the APR of holding the token can increase. Long-term holders will receive rewards and short-term holders/traders would not receive the same benefit.
+
+
```

Comment:
Another possible appliance is to reward user for lending or borrowing in lending protocols

Reply:
fixed